containers/daap-python-base/src/var/task/versioning.py

```diff
@@ -314,6 +315,15 @@ def update_schema(self, input_data: dict, table_name: str) -> tuple[str, dict]:
             )
             schema.convert_schema_to_glue_table_input_csv()
             schema.write_json_to_s3(new_version_key)
+            # if major we need to create next major version data product data
```

Comment:
I've been wondering about whether this version update operation needs to be treated as an atomic operation. I.e. if something fails part way through, should we attempt to roll back the metadata?

I think this is probably something that is OK for now, but we could make it more resilient later on (i.e. once we have users!)